src/palace.pyx

```diff
@@ -85,6 +85,16 @@ channel_configs: FrozenSet[str] = frozenset({
     'B-Format 2D', 'B-Format 3D'})
 
 
+def current_context() -> Optional[Context]:
+    """The context that is currently used."""
+    dev = Device(None)
+    current_impl = alure.Context.get_current()
+    dev.impl = current_impl.get_device()
+    current_ctx = Context(dev)
+    current_ctx.impl = current_impl
```

Comment:
I don't think I've made my idea clear. What I mean is some thing like this:
```python
cdef alure.Context current_impl = alure.Context.get_current()
cdef Context current = Context(None) # this needs tweaks to work too; it should avoid creating a new alure.Context
current.impl = current_impl
current.device = Device(None)
current.device.impl = current_impl.get_device() # don't forget exposing this C++ method
current.listener = Listener(current)
return current
```

Reply:
The problem with `Context(None)` is because you can't initialize a context with `None`. In the constructor of `Context`:

```python
def __init__(self, device: Device, attrs: Dict[int, int] = {}) -> None:
    self.impl = device.impl.create_context(mkattrs(attrs.items()))
```
If `device=None` then it would raise an error because you're calling `create_context` from `None`.

```bash
>   self.impl = device.impl.create_context(mkattrs(attrs.items()))
E   RuntimeError: alcCreateContext failed: Invalid device
```

Reply:
> The problem with `Context(None)` is because you can't initialize a context with `None`.

That's right, hence we won't *initialize*:

> cdef Context current = Context(None) # this needs tweaks to work too; it should avoid creating a new alure.Context

Speaking of the idiom, this is what I mean:
```python
cdef class Device:
    """...
    Parameters
    ----------
    name : Optional[str], optional
        The name of the playback device.  If it is `None`,
        the object is left uninitialized.
    ...
    """
    cdef alure.Device impl
    def __init__(self, name: Optional[str] = '',
                 fail_safe: bool = False) -> None:
        if name is None: return
        ...
```
It's not very pretty tho, best way is to call `__new__`, you can look up the Cython doc on what it does on extension types and see if that works.  If it does, we'll refactor the current code using the `None` idioms but that's not within the scope of this issue.

Reply:
I mean I copied the code above and got the error above 🤷‍♂️

Reply:
Have you did what the comment asked (tweaking the Context class)?  BTW the edge case is forgotten.

Reply:
What does "tweak" mean?

Reply:
I'm not sure, [looking it up](https://lmddgtfy.net/?q=!%20tweak) might help.  BTW I've referenced [the docs](https://docs.python.org/3/reference/datamodel.html#object.__new__) and seemingly you can avoid doing the tweak and go for `cdef Context current = Context.__new__(Context)`.